.github/copilot-instructions.md

```diff
@@ -0,0 +1,168 @@
+# GitHub Copilot Instructions for Azure TRE
+
+This file provides context and guidance for GitHub Copilot when working with the Azure Trusted Research Environment (Azure TRE) project.
+
+## Project Overview
+
+Azure TRE is an accelerator to assist Microsoft customers and partners who want to build out Trusted Research Environments on Azure. It enables authorized users to deploy and configure secure workspaces and researcher tooling without a dependency on IT teams.
+
+Trusted Research Environments (TREs) enable organizations to provide research and development (R&D) teams secure access to data alongside tooling to ensure productivity while keeping security controls in place.
+
+Core features include:
+- Self-service workspace management for TRE administrators
+- Self-service provisioning of R&D tooling for R&D teams
+- Package and repository mirroring
+- Extensible architecture with custom service templates
+- Microsoft Entra ID integration
+- Airlock for import and export
+- Cost reporting
+
+## Primary Technologies
+
+Azure TRE uses the following key technologies:
+
+- **Infrastructure as Code**:
+  - Terraform for infrastructure provisioning
+  - Porter/CNAB for bundle packaging
+
+- **Languages**:
+  - Python (API, resource processor)
+  - TypeScript/JavaScript (UI)
+  - Bash (deployment scripts)
+  - HCL (Terraform)
+  - YAML (CI/CD pipelines, Porter manifests)
+
+- **Cloud Services**:
+  - Azure services (App Service, Container Registry, Cosmos DB, etc.)
+  - Microsoft Entra ID for authentication
+  - Azure VMSS for resource processor
+
+- **Development Tools**:
+  - Docker for containerization
+  - GitHub Actions for CI/CD
+  - Make for build/deployment automation
```

Comment:
We also have our own tre cli. Which is placed under the cli folder and allows us to run common commands needed for tre setup, auth and etc...